src/main/scala/firrtl/passes/InferWidths.scala

```diff
@@ -229,6 +229,7 @@ object InferWidths extends Pass {
       case (t1: SIntType, t2: SIntType) => Seq(WGeq(t1.width, t2.width))
       case (ClockType, ClockType) => Nil
       case (FixedType(w1, p1), FixedType(w2, p2)) => Seq(WGeq(w1,w2), WGeq(p1,p2))
+      case (AnalogType(w1), AnalogType(w2)) => Seq(WGeq(w1,w2))
```

Comment:
The semantics of attach are that all wires must have exactly the same width, because there aren't sign/zero-extend semantics for mismatched widths.